### PR TITLE
pkgs/applications: Fix some unstableGitUpdater users

### PR DESCRIPTION
**Pull request by OPNA2608**
## Description of changes

#280501 now makes `unstableGitUpdater` produce versions that match our desired unstable version schema. Start fixing the versions of packages that use it, and their `unstableGitUpdater` arguments as needed.

Excluded from here are `nixops_unstable.passthru.plugins` and `mpvScripts` attrs, just to keep the scope abit more managable.

Pinging @eclairevoyant since they seem to have some interest in implementing this more (sorry if I'm wrong about that!).

---

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>28 packages built:</summary>
  <ul>
    <li>_64gram</li>
    <li>chirp</li>
    <li>chirp.dist</li>
    <li>eigenmath</li>
    <li>kile-wl</li>
    <li>libtransmission_4</li>
    <li>libtransmission_4.apparmor</li>
    <li>libutp_3_4</li>
    <li>littlegptracker</li>
    <li>miriway</li>
    <li>mmlgui</li>
    <li>mopidy-spotify</li>
    <li>mopidy-spotify.dist</li>
    <li>retroarch</li>
    <li>retroarch-assets</li>
    <li>retroarchFull</li>
    <li>telegram-desktop</li>
    <li>transgui</li>
    <li>transmission_4</li>
    <li>transmission_4-gtk</li>
    <li>transmission_4-gtk.apparmor</li>
    <li>transmission_4-qt</li>
    <li>transmission_4-qt.apparmor</li>
    <li>transmission_4-qt6</li>
    <li>transmission_4-qt6.apparmor</li>
    <li>transmission_4.apparmor</li>
    <li>vgmplay-libvgm</li>
    <li>wdt</li>
  </ul>
</details>

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/emulators/retroarch/retroarch-assets.nix [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 stdenvNoCC.mkDerivation {
   pname = "retroarch-assets";
-  version = "unstable-2024-04-24";
+  version = "0-unstable-2024-04-24";
```
Comment (eclairevoyant) [inline pkgs/applications/emulators/retroarch/retroarch-assets.nix:9]:
```suggestion
  version = "1.17.0-unstable-2024-04-24";
```

seems like this has a proper tag?
https://github.com/libretro/retroarch-assets/releases/tag/v1.17.0

File: pkgs/applications/emulators/retroarch/retroarch-assets.nix [hunk 2]
```diff
@@ -24,7 +24,10 @@ stdenvNoCC.mkDerivation {
 
   dontBuild = true;
 
-  passthru.updateScript = unstableGitUpdater { };
+  passthru.updateScript = unstableGitUpdater {
+    # Only has a "Latest" tag
+    hardcodeZeroVersion = true;
```
Comment (eclairevoyant) [inline pkgs/applications/emulators/retroarch/retroarch-assets.nix:29]:
```suggestion
    tagPrefix = "v";
```

**Review by eclairevoyant — COMMENTED** [review]
other packages match what I'd expect

File: pkgs/applications/emulators/retroarch/retroarch-assets.nix (hunk 1 above)
Comment (OPNA2608) [reply pkgs/applications/emulators/retroarch/retroarch-assets.nix]: Oops, yup. Fixed.

**Review by eclairevoyant — APPROVED** [review]

**Comment by OPNA2608** [comment]
(Are you interested in being pinged on subsequent PRs like this @eclairevoyant? Checking so I don't needlessly shower your inbox. :sweat_smile:)

**Comment by eclairevoyant** [comment]
Sure.

**Pull request merged**